### PR TITLE
Add pyyaml

### PR DESCRIPTION
**Pull request by jakirkham**
Adds a recipe to build `pyyaml`. Generated with `conda skeleton pypi` and made some tweaks.

If anyone has an interest in being a maintainer on this or `yaml`, please let me know and I will happily add you. Also, please let me know if you have any thoughts on this recipe.

cc @pelson @ocefpaf @msarahan @mcg1969

### CODE REVIEW

**Comment by msarahan** [comment]
Go ahead and add me as maintainer if you want.  Do you want more tests than a simple import test?

Otherwise, LGTM.

**Comment by jakirkham** [comment]
Thanks @msarahan. Added you above.

Also, I was going to make some tweaks that would include building the C-bindings, which are not currently handled. With that I will include running the test suite during the build.

**Comment by jakirkham** [comment]
Do you know why it cannot find `yaml.h` on Windows?

**Comment by msarahan** [comment]
where do you see that?  Looks to me like things are working fine?

**Comment by jakirkham** [comment]
I don't think it is all of them that have this issue, but this [one]( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.2884/job/stoavijkomrn7yom#L478 ) certainly does.

**Comment by jakirkham** [comment]
Actually, it is only that one build. So, on Windows Python 3.4 32-bit we are seeing this issue. I'll add a test to make this more obvious.

**Comment by jakirkham** [comment]
Also, kind of interesting, see this [line]( https://github.com/conda/conda-recipes/blob/0dabe789218ef58df023154dfc6482791b336451/python/pyyaml/meta.yaml#L21 ) in conda-recipes.

**Comment by msarahan** [comment]
That's VS 2010, which is the weird one with activation.  I bet it's overwriting the INCLUDE variable.

**Comment by jakirkham** [comment]
It looks like its include flags for the compiler look ok. See this [line]( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.2885/job/anjkd85letr050tw#L504 ).

**Comment by msarahan** [comment]
Well, I'm out of suggestions.  I manually downloaded the package for VS 2010, and it does contain the right files.

**Comment by pelson** [comment]
👍 - LGTM

**Comment by jakirkham** [comment]
> Well, I'm out of suggestions. I manually downloaded the package for VS 2010, and it does contain the right files.

Which one? The one from `defaults` or the one here?

**Comment by jakirkham** [comment]
So, given that Windows Python 3.4 64-bit was passing and that is pinned to `conda-build` 1.20.0, I thought let's try extending that pinning to Windows Python 3.4 32-bit and see what happens. Sure enough that fixes the problem here. In short, it looks like we may be hitting a `conda-build` bug. Want to take a look, @msarahan?

If there is no opposition, I'll revert the CI change and merge. We can then add the workaround to the feedstock. Sound good?

**Comment by jakirkham** [comment]
Reversion has been pushed. CI will not be changed at staged-recipes.

This will now fail again on that one Windows build, but we now know to fix it. So, it should not be a blocker for merging.

**Comment by jakirkham** [comment]
Going ahead with this. Will add workaround in the feedstock.

**Comment by jakirkham** [comment]
Opened an issue ( https://github.com/conda/conda-build/issues/993 ) about the build problem we are encountering.

File: recipes/pyyaml/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+source:
+  fn: PyYAML-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/P/PyYAML/PyYAML-{{ version }}.tar.gz
+  md5: f50e08ef0fe55178479d3a618efe21db
+
+build:
+  number: 0
+  script:
+    - python setup.py --with-libyaml build_ext --include-dirs="${PREFIX}/include" --library-dirs="${PREFIX}/lib"  # [unix]
+    - python setup.py --with-libyaml build_ext --include-dirs="%LIBRARY_INC% --library-dirs=%LIBRARY_LIB%         # [win]
```
Comment (msarahan) [inline recipes/pyyaml/meta.yaml]: The error with Appveyor is a mismatched quote here, not a bug in Conda-build.  ```"%LIBRARY_INC```

I'm leaving the test in, but can no longer reproduce the failure.

Comment (jakirkham) [inline recipes/pyyaml/meta.yaml]: Oops. Sorry. 😳 Will fix that. Thanks for spotting that.

Comment (jakirkham) [inline recipes/pyyaml/meta.yaml]: See this PR ( https://github.com/conda-forge/pyyaml-feedstock/pull/4 ).

Comment (jakirkham) [inline recipes/pyyaml/meta.yaml]: Though I must say I am a bit surprised that this only caused issues with `conda-build` on that one version of Python and went away with a downgrade. Is there some reason it didn't show up in the other cases?

Comment (msarahan) [inline recipes/pyyaml/meta.yaml]: I am mystified.  I think the main reason is that it is indeed a setting of INCLUDE going wrong (setenv.cmd replacing it), but the others set INCLUDE correctly (and the --include-dirs is superfluous there).  I'm guessing it is only strictly necessary for the VS 2010 build?  Wild conjecture here.

**Pull request merged**